nixos/modules/services/networking/create_ap.nix

```diff
@@ -0,0 +1,160 @@
+{ config, lib, pkgs, ... }:
+with lib;
+
+let
+  cfg = config.services.create_ap;
+  configFile = pkgs.writeText "create_ap.conf" ''
+    CHANNEL=${toString cfg.channel}
+    GATEWAY=${cfg.gateway}
+    WPA_VERSION=2
+    ETC_HOSTS=0
+    DHCP_DNS=gateway
+    NO_DNS=0
+    NO_DNSMASQ=0
+    HIDDEN=0
+    MAC_FILTER=0
+    MAC_FILTER_ACCEPT=/etc/hostapd/hostapd.accept
+    ISOLATE_CLIENTS=0
+    SHARE_METHOD=nat
+    IEEE80211N=0
+    IEEE80211AC=0
+    HT_CAPAB=[HT40+]
+    VHT_CAPAB=
+    DRIVER=nl80211
+    NO_VIRT=0
+    COUNTRY=
+    FREQ_BAND=2.4
+    NEW_MACADDR=
+    DAEMONIZE=0
+    NO_HAVEGED=0
+    WIFI_IFACE=${cfg.wifi-iface}
+    INTERNET_IFACE=${cfg.internet-iface}
+    SSID=${cfg.ssid}
+    PASSPHRASE=${cfg.passphrase}
+    USE_PSK=0
+    ${cfg.extraConfig}
```

Comment:
Now that I've opened NixOS/rfcs#42, I'm not so sure anymore about this anymore. Because once we introduce such hardcoded defaults and `extraConfig`, it's hard to go back. Consider using the approach in the RFC instead.